### PR TITLE
Drop support for Hbase 1

### LINKED ISSUE #2212: Drop support HBase 1
**Describe the feature:**
Drop support HBase 1. It is required to drop hbase 1 since it requires a quite old version guava and we support also for a longer time now hbase 2.

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #2212

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by farodin91** [comment]
@mbrukman Do you like to check if bigtable is correctly after these changes?

**Comment by farodin91** [comment]
@pluradj @FlorianHockmann Do you like to review it?

**Comment by porunov** [comment]
@farodin91 I see that we have tests for hbase1 which are passing. If so, does it mean that hbase1 works with JanusGraph? Also, what if we exclude guava and add an older version of guava into classpath along with the JanusGraph libs which are depending on never guava version. Will that break something or will it fix hbase1 (in case it doesn't work due to guava).
I'm just trying to understand if we really need to remove hbase1 support or can we just add a documentation which says `to work with hbase1 you need to use an older guava version. I.e. exclude newer version from janusgraph and add an older version`. 

So, in short I have 2 broad questions:
1. Are we sure thay hbase1 doesn't work with JanusGraph?
2. Can we add documentation about hbase1 instead of removing hbase1 support?

**Comment by porunov** [comment]
@farodin91 should we move this PR to later releases or do we need to resolve it in 0.6.0?

**Comment by farodin91** [comment]
I don't think it should be released blocking.

**Comment by porunov** [comment]
@farodin91 I don't think I fully understand why we need to drop support for HBase 1. As I said, we have hbase 1 tests right now in master and they are passing.
Also, if there are any tests (which I'm not aware of) which are not passing or we know that hbase 1 doesn't work with JanusGraph because of newer guava version, we can just say in documentation that users should decrease their guava version to necessary guava version which is supported by hbase 1.
In case hbase 1 doesn't work with JanusGraph even after guava version decreese, then we need to drop support for hbase 1 (as you do in this PR).

**Comment by farodin91** [comment]
I have to say I'm not 100% sure if hbase 2 or hbase 1 which is currently tested. I think maven profile are a bit _confusing_ for hbase. Release builds already just support one version. 

Support matrix between clients and server versions is hard to find. (It seems client and server are hardly bounded.) I can also not say if hbase1 is supported by Apache and we want to support only supported version of backends. 

Their is already alpha version of hbase3. (This doesn't tell anything about a new release in the near future.)

Side note: I removed all usage of deprecated client functions. 

I tried to test hbase1 against the newest version which runs into a timeout how knows why.

**Comment by farodin91** [comment]
> Side note: I removed all usage of deprecated client functions.

I checked the hbase those replacement function are added in hbase 2, so they don't exists in hbase 1.

**Comment by porunov** [comment]
> I can also not say if hbase1 is supported by Apache and we want to support only supported version of backends.

Hbase 1 is supported. 20 days ago there was a release candidate `1.7.0RC0`.

**Comment by porunov** [comment]
To be clear, I'm not using hbase 1 personally and I'm OK if we need to drop support for this storage backend. I just didn't fully understand the reason behind dropping hbase 1 storage backend.
The only reason I see in the issue is:
>  Drop support HBase 1. It is required to drop hbase 1 since it requires a quite old version guava and we support also for a longer time now hbase 2.

Thus, I had thoughts that if we use older guava version is those installs which  require hbase 1, we could still have support for hbase 1. If that's not the case, I'm totally fine dropping the support for this backend.
I just think that we need proceed with this PR as soon as possible as this is the only blocking PR for JanusGraph 0.6.0 release (at least that's what I understand).

**Comment by farodin91** [comment]
@porunov I will extract the all changes outside of drop and I will try to cleanup the maven stuff as much as possible. What do you think?

**Comment by porunov** [comment]
> @porunov I will extract the all changes outside of drop and I will try to cleanup the maven stuff as much as possible. What do you think?

Sounds good to me. Thank you!

**Comment by farodin91** [comment]
@porunov I will currently try to get hbase 1 to running in different branch. It looks like it don't work. 

I will check as next a older commit before (testcontainers).

**Comment by farodin91** [comment]
@porunov Tests doesn't work for long with hbase1. Their was an issue in the pom, which prevent the execution of correct hbase1 tests. I'm happy to get rid of before 1.0, if it is not really working in our test cases. 

What do you think?

**Comment by porunov** [comment]
I personally don't use HBase 1 and I don't know if anyone in the community is using HBase 1. I guess, it would make sense to start a thread in `janusgraph-dev` mailing list to see to warn community of this change. Maybe we could get some feedback in this situation. 
I personally totally fine dropping support for HBase 1 and support HBase 2 only.

As a side note, TinkerPop bumped Hadoop versions to 3.x. I didn't check if HBase 1 can run on Hadoop 3. I didn't check the compatibility overall but I assume there could potentially be some incompatibilities (again, didn't check). 
I also remember that a while ago we had support for Hadoop 1 and Hadoop 2. Right now JanusGraph has support for Hadoop 2 but we might need to add support for Hadoop 3 when upgrading TinkerPop to 3.6.0 version (unreleased yet).

**Comment by farodin91** [comment]
@porunov  I ask on the user list to get a bit more feedback.

**Comment by porunov** [comment]
> @porunov I ask on the user list to get a bit more feedback.

Just link the mailing thread to this PR, so that we would have a reference to the mailing thread.

**Comment by farodin91** [comment]
@porunov it doesn't look like a lot of people are using hbase 1. 

I will check the current build failures and will rebase the PR.

**Comment by porunov** [comment]
> I will check the current build failures and will rebase the PR.

Agree. I believe it's OK to remove support for HBase 1.

> I will check the current build failures and will rebase the PR.

I don't know what happened but dist tests are now constantly failing (tried to restart several times). It started to happen from this commit: https://github.com/JanusGraph/janusgraph/runs/3655379551

I didn't have a chance to look at it yet.

**Comment by farodin91** [comment]
@porunov  It looks like the cache is broken.

**Comment by farodin91** [comment]
@porunov build is now passing. 🎉

File: .github/workflows/ci-backend-hbase.yml [hunk 1]
```diff
@@ -62,25 +62,13 @@ jobs:
...
-            name: hbase2-diskstorage
-          - module: hbase
-            install-args: "-Dhbase.profile -Phbase2"
-            args: "-Dtest=\"**/graphdb/hbase/*\""
-            name: hbase2-graphdb
-          - module: hbase
-            install-args: "-Dhbase.profile -Phbase2"
-            args: "-Dtest=\"**/hadoop/*\""
-            name: hbase2-hadoop
+            name: hbase-hadoop
```
Comment (porunov) [inline .github/workflows/ci-backend-hbase.yml:71]: I'm not sure yet as I didn't read about HBase compatibility but if we decide to support HBase 3 after it becomes GA, will we have `hbase-hadoop` and `hbase3-hadoop`? If so I think it would make sense to have `hbase2-hadoop` and `hbase3-hadoop`.
Currently HBase 3 is in Alpha: https://github.com/apache/hbase/releases/tag/rel%2F3.0.0-alpha-1
That said, in case we can use HBase 3 client libraries to work with HBase to than it totally makes sense to have just one module `hbase-hadoop`.
I also didn't check if we can use hadoop3 libs to work with hadoop2. Right now JanusGraph supports Hadoop 2 because of TinkerPop but TinkerPop bumps Hadoop to 3 version in 3.6.0-SNAPSHOT version. Thus, I think JanusGraph will need to support both Hadoop 3 and Hadoop 2. In this case we just need to understand compatibility because we will have:
Hadoop 2
Hadoop 3
HBase 2
HBase 3

You can ignore this comment, it's just something we will need to think about later

File: docs/configs/janusgraph-cfg.md [hunk 2]
```diff
@@ -514,7 +514,6 @@ HBase storage options
 
 | Name | Description | Datatype | Default Value | Mutability |
 | ---- | ---- | ---- | ---- | ---- |
-| storage.hbase.compat-class | The package and class name of the HBaseCompat implementation. HBaseCompat masks version-specific HBase API differences. When this option is unset, JanusGraph calls HBase's VersionInfo.getVersion() and loads the matching compat class at runtime.  Setting this option forces JanusGraph to instead reflectively load and instantiate the specified class. | String | (no default value) | MASKABLE |
```
Comment (porunov) [inline docs/configs/janusgraph-cfg.md:517]: I guess we just need to double check that we won't need this configuration when we bring support for HBase 3

**Review by porunov — COMMENTED** [review]
I didn't check all the changes yet but I'm just thinking if we need to leave `hbase2` instead of changing it to `hbase`. I might be strange if we have `hbase` and `hbase3` modules instead of `hbase2` and `hbase3` but to check weather we really need 2 modules or not we need to check hbase 2 and hbase 3 compatibility I guess

File: docs/configs/janusgraph-cfg.md (hunk 2 above)
Comment (farodin91) [reply docs/configs/janusgraph-cfg.md:517]: I don't think as they say client should compatible: https://github.com/apache/hbase/blob/master/src/main/asciidoc/_chapters/upgrading.adoc#hbase.rolling.upgrade

File: .github/workflows/ci-backend-hbase.yml (hunk 1 above)
Comment (farodin91) [reply .github/workflows/ci-backend-hbase.yml:71]: I think it is a good idea to mark the tests as hbase2-... .

**Comment by farodin91** [comment]
@porunov Would you like check this PR again?

**Comment by farodin91** [comment]
@porunov  Otherwise, i would merge it in one or two weeks.

File: pom.xml [hunk 3]
```diff
@@ -1216,6 +1215,16 @@
...
+            <dependency>
+                <groupId>org.apache.htrace</groupId>
+                <artifactId>htrace-core4</artifactId>
+                <version>4.2.0-incubating</version>
+            </dependency>
+            <dependency>
+                <groupId>javax.enterprise</groupId>
+                <artifactId>cdi-api</artifactId>
+                <version>1.0-SP4</version>
+            </dependency>
```
[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml:1227]: I will check

[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml:1227]: Otherwise the dependency checker would fail.

[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml:1227]: We can remove it in the future, if update hbase and hadoop.

[the remark this answers is not in the document]
Comment (porunov) [reply pom.xml:1227]: Can we move the added dependencies into `janusgraph-hbase/pom.xml` or do you think they should be included into other modules as well?

[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml:1227]: It won't be include. I will move it.

[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml]: Moved

[the remark this answers is not in the document]
Comment (farodin91) [reply pom.xml]: I moved also htrace-core4

**Review by porunov — APPROVED** [review]
LGTM. Thank you @farodin91 !
